test/SlowTests/Issues/RavenDB-23380.cs

```diff
@@ -151,7 +155,41 @@ private void InvalidDefinitionShouldThrow(Options options, bool isSharded)
             }
         }
     }
+
+    [RavenFact(RavenTestCategory.Indexes)]
+    public async Task TestIndex_WhenIndexNameIsSubstringOfQuery_ShouldTranslateTestIndexParametersCorrectly()
+    {
+        const int expectedValue = 9;
+        
+        var payload = new TestIndexParameters
+        {
+            IndexDefinition = new IndexDefinition { Name = "i", Maps = new HashSet<string> { "from obj in docs.Dtos select new { IndexProp = obj.Value }" } },
+            Query = "from index 'i'"
```

Comment:
`Query = "from index \"i\""`

That would break the previous code

Reply:
Is that a question?
It will be replaced by the temp name so the result will be "from {tempName}ndex \"{tempName}\""